### PR TITLE
Remove dependencies from `cardano-ledger-alonzo` and `cardano-ledger-byron`

### LINKED ISSUE #608: [FEATURE] - Documentation and Integration Clarification
**Internal/External** : *Internal* 
## Context  
The current repository lacks documentation, making it unclear what direction Intersect is taking with it.

When the repository is updated, projects that depend on it face difficulties upgrading due to the following reasons, among others:
  - `cardano-node` bypasses the dependencies `cardano-api` is supposed to abstract. 
  -  Similar concepts but from different core domains are mixed : e.g : `Ledger.Coin` and `Cardano.API.Lovelace` (that has been removed in 8.40 and not explained in the changelog > see #607 )
 
Here is the current documentation of the Readme :
 
image

I am requesting the addition of comprehensive documentation to assist projects relying on this dependency in progressing with their own development.

## What is the "vision" of this repository?

The repository mentions the **integration of the [ledger](https://github.com/IntersectMBO/cardano-ledger), [consensus](https://github.com/IntersectMBO/ouroboros-consensus), and [networking](https://github.com/IntersectMBO/ouroboros-network/tree/master/ouroboros-network) repositories**.

Is `cardano-api` intended to be the interface for handling Cardano-related tasks? If so, why do some projects use `cardano-api` but also directly access the repositories behind the API?

I was expecting to see something more cohesive, like this:
image

But I can see that most of the repositories using this API are bypassing it : 
### Cardano Node and Cardano Submit API
image
### Cardano CLI
image

These issues impact projects built on top of these libraries. I believe it starts with a clarification on the project vision. What is `cardano-api` aimed to be ?

#### Repository Vision Detailed

The README should clearly outline the vision of the repository, including its purpose and long-term goals. This section should also clarify the intended role of `cardano-api` in handling Cardano-related tasks, addressing why some projects might use both `cardano-api` and the underlying repositories directly.

The current `cardano-node` documentation links should be removed , it is irrelevant here.

#### High-Level Features and Examples of Code

The README should include:
- A detailed description of the features the repository aims to provide.
- Code examples demonstrating how to use these features.
- Information on where this repository is used (e.g., cardano-node, cardano-cli, cardano-submit-api).

#### High-Level Issues and Roadmap/Backlog

Please provide details on:
- The current focus and ongoing work of the core maintainers.
- The roadmap and backlog for future development.

Adding comprehensive documentation will help projects relying on this dependency better understand its capabilities and future direction, facilitating smoother development and integration.

### PR DESCRIPTION
**Pull request by palas**
### Note: must remove SRP before merging

# Changelog

```yaml
- description: |
    Removed dependencies from `cardano-ledger-alonzo` and `cardano-ledger-byron`
  type:
  - refactoring    # QoL changes
```

# Context

As part of the work targeted at addressing issues pointed out in https://github.com/IntersectMBO/cardano-api/issues/608, this PR aims to remove all direct dependencies from ledger (in this PR `alonzo` and `byron` parts) by re-exposing them through `cardano-api`,

# How to trust this PR

Make sure it is just a refactoring.

Look at it in conjunction with: https://github.com/IntersectMBO/cardano-api/pull/647

Take advantage that there are one commit for alonzo and one for byron (same for the corresponding PR in `cardano-api`)

# Checklist

- [x] Commit sequence broadly makes sense and commits have useful messages
- [x] New tests are added if needed and existing tests are updated. See [Running tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [x] Self-reviewed the diff

### CODE REVIEW

File: cardano-cli/src/Cardano/CLI/EraBased/Run/Genesis.hs [hunk 1]
```diff
@@ -404,28 +396,28 @@ runGenesisCreateCardanoCmd
     let
       byronGenesis =
         byronGenesis'
-          { gdProtocolParameters =
-              (gdProtocolParameters byronGenesis')
-                { ppSlotDuration = floor (toRational slotLength * recip slotCoeff)
+          { L.gdProtocolParameters =
```
Comment (Jimbo4350) [inline cardano-cli/src/Cardano/CLI/EraBased/Run/Genesis.hs]: So everything Byron related should be exposed from `Cardano.Api.Byron`. Everything else looks good in this PR :+1:

File: cardano-cli/src/Cardano/CLI/Byron/Commands.hs [hunk 2]
```diff
@@ -104,12 +102,12 @@ data NodeCmds
   | UpdateProposal
       NetworkId
       (SigningKeyFile In)
-      ProtocolVersion
-      SoftwareVersion
-      SystemTag
-      InstallerHash
+      Byron.ProtocolVersion
```
Comment (Jimbo4350) [inline cardano-cli/src/Cardano/CLI/Byron/Commands.hs:105]: :+1:

File: cardano-cli/src/Cardano/CLI/Byron/Delegation.hs [hunk 3]
```diff
@@ -16,8 +16,6 @@ where
 import           Cardano.Api.Byron
 import qualified Cardano.Api.Ledger as L
```
Comment (Jimbo4350) [inline cardano-cli/src/Cardano/CLI/Byron/Delegation.hs:17]: What are we exporting from `Cardano.Api.Ledger`?

File: cardano-cli/src/Cardano/CLI/Byron/Genesis.hs [hunk 4]
```diff
@@ -15,12 +15,8 @@ where
 import           Cardano.Api (Key (..), NetworkId, writeSecrets)
```
Comment (Jimbo4350) [inline cardano-cli/src/Cardano/CLI/Byron/Genesis.hs:15]: Not for this PR but the fact we are exposing `Cardano.Api` here could mean byron related things are being exported via `Cardano.Api`

**Review by Jimbo4350 — APPROVED** [review]
Good work :+1: . There are still instances where `Cardano.Api.Byron` and `Cardano.Api` are being imported. We need to figure out if we can remove the `Cardano.Api` imports.

**Review by carbolymer — APPROVED** [review]

File: cardano-cli/src/Cardano/CLI/Byron/Delegation.hs (hunk 3 above)
Comment (palas) [reply cardano-cli/src/Cardano/CLI/Byron/Delegation.hs:17]: Apparently these three only:
- [`Annotated(..)`](https://cardano-ledger.cardano.intersectmbo.org/cardano-ledger-binary/Cardano-Ledger-Binary-Decoding.html#t:Annotated)
- [`serialize'`](https://cardano-ledger.cardano.intersectmbo.org/cardano-ledger-binary/Cardano-Ledger-Binary-Encoding.html#v:serialize-39-)
- [`byronProtVer`](https://cardano-ledger.cardano.intersectmbo.org/cardano-ledger-binary/Cardano-Ledger-Binary-Version.html#v:byronProtVer)

**Comment by palas** [comment]
Solved by https://github.com/IntersectMBO/cardano-cli/pull/936/commits/105d59688d2323334e2fbb7d4c9da7ebeb3e5354

**Pull request closed without merging**